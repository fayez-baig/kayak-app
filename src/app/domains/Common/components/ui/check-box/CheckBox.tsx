import { FC, ReactElement } from "react";
import { CheckBoxProps } from "./types";
import "./checkbox.scss";

const CheckBox: FC<CheckBoxProps> = ({
  onChange,
  value,
  label,
  name,
  checked,
  ...props
}): ReactElement => {
  return (
    <div className="checkbox-wrapper">
      <input
        type="checkbox"
        id={name}
        name={name}
        checked={checked}
        data-testid={name}
        value={value}
        onChange={onChange}
        {...props}
      />
      <label htmlFor={name}>{label}</label>
    </div>
  );
};

export default CheckBox;
