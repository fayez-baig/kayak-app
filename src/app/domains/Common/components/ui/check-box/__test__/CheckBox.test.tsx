import { createRenderer } from "react-test-renderer/shallow";
import CheckBox from "../CheckBox";

const renderer = createRenderer();

const props = {
  onChange: jest.fn(),
  value: "OW",
  label: "Oneworld",
  name: "Oneworld",
  checked: true,
};

describe("<CheckBox />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(<CheckBox {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
