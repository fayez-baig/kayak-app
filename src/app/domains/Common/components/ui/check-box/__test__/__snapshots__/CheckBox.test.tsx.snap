// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should render and match the snapshot 1`] = `
<div
  className="checkbox-wrapper"
>
  <input
    checked={true}
    data-testid="Oneworld"
    id="Oneworld"
    name="Oneworld"
    onChange={[MockFunction]}
    type="checkbox"
    value="OW"
  />
  <label
    htmlFor="Oneworld"
  >
    Oneworld
  </label>
</div>
`;
